lib/rules/no-wait-for-side-effects.ts

```diff
@@ -52,18 +60,31 @@ export default createTestingLibraryRule<Options, MessageIds>({
       body: TSESTree.Node[]
     ): TSESTree.ExpressionStatement[] {
       return body.filter((node) => {
-        if (!isExpressionStatement(node)) {
+        if (!isExpressionStatement(node) && !isVariableDeclaration(node)) {
           return false;
         }
 
         const expressionIdentifier = getPropertyIdentifierNode(node);
-        if (!expressionIdentifier) {
-          return false;
-        }
+
+        const isRenderInVariableDeclaration =
+          isVariableDeclaration(node) &&
+          node.declarations.some((declaration) =>
+            helpers.isRenderVariableDeclarator(declaration)
+          );
```

Comment:
I think early returns are cleaner
```suggestion
        const isRenderInVariableDeclaration =
          isVariableDeclaration(node) &&
          node.declarations.some((declaration) =>
            helpers.isRenderVariableDeclarator(declaration)
          );
          if (!isRenderInVariableDeclaration) {
            return false;
          }
```

Reply:
Early returns wouldn't work here, because we need to account for `helpers.isFireEventUtil(expressionIdentifier)` and other cases. Am I missing something?

Reply:
As you're using `||` instead of `&&`, you're probably right for this function

Reply:
We could have an early return when the clause is `true`. Do you think that's better?

```
    function getSideEffectNodes(
      body: TSESTree.Node[]
    ): TSESTree.ExpressionStatement[] {
      return body.filter((node) => {
        if (!isExpressionStatement(node) && !isVariableDeclaration(node)) {
          return false;
        }

        const isRenderInVariableDeclaration =
          isVariableDeclaration(node) &&
          node.declarations.some((declaration) =>
            helpers.isRenderVariableDeclarator(declaration)
          );

        if (isRenderInVariableDeclaration) {
          return true;
        }

        const isRenderInAssignment =
          isExpressionStatement(node) &&
          isAssignmentExpression(node.expression) &&
          helpers.isRenderUtil(
            getPropertyIdentifierNode(node.expression.right)
          );

        if (isRenderInAssignment) {
          return true;
        }

        const expressionIdentifier = getPropertyIdentifierNode(node);

        return (
          helpers.isFireEventUtil(expressionIdentifier) ||
          helpers.isUserEventUtil(expressionIdentifier) ||
          helpers.isRenderUtil(expressionIdentifier)
        );
      }) as TSESTree.ExpressionStatement[];
    }
```

Reply:
That's maybe a good idea indeed

Just to not make the final return too big and making the cognitive load lower when looking at the code.

Reply:
Extracting the calculations into a separate function can help too for readability btw 🤔

And I think I prefer to do it that way actually 🤔

CC/ @Belco90